rasa/shared/core/slot_mappings.py

```diff
@@ -167,6 +167,39 @@ def entity_is_desired(
 
         return slot_fulfils_entity_mapping
 
+    @staticmethod
+    def check_mapping_validity(mapping: Dict[Text, Any], domain: "Domain") -> bool:
+        """Checks the mapping for validity.
+
+        Args:
+            mapping: Slot mapping.
+            domain: The domain to check against.
+
+        Returns:
+            True, if intent and entity specified in a mapping exist in domain.
+        """
+        from rasa.shared.core.constants import MAPPING_TYPE
+
+        if (
+            mapping.get(MAPPING_TYPE) == str(SlotMapping.FROM_ENTITY)
+            and mapping.get(ENTITY_ATTRIBUTE_TYPE) not in domain.entities
+        ):
+            rasa.shared.utils.io.raise_warning(
+                f"Slot uses a 'from_entity' mapping for a non-existent entity '{mapping.get(ENTITY_ATTRIBUTE_TYPE)}'"
```

Comment:
```suggestion
                f"Slot uses a 'from_entity' mapping for a non-existent entity '{mapping.get(ENTITY_ATTRIBUTE_TYPE)}' "
                f"Skipping slot extraction because of invalid mapping."
```